7.7.1/stretch/Dockerfile

```diff
@@ -0,0 +1,92 @@
+FROM debian:stretch-slim
+LABEL maintainer "Dave Curylo <dave@curylo.org>, Michael Hendricks <michael@ndrix.org>"
+RUN apt-get update && \
+    apt-get install -y --no-install-recommends \
+    libarchive13 \
+    libgmp10 \
+    libossp-uuid16 \
+    libssl1.1 \
+    ca-certificates \
+    libdb5.3 \
+    libpcre3 \
+    libedit2 \
+    libgeos-c1v5 \
+    libspatialindex4v5 \
+    unixodbc \
+    odbc-postgresql \
+    tdsodbc \
+    libmariadbclient18 \
+    libsqlite3-0 \
+    librocksdb4.5 \
+    libserd-0-0 \
+    libraptor2-0 && \
+    rm -rf /var/lib/apt/lists/*
+RUN SWIPL_VER=7.7.1 \
+    SWIPL_CHECKSUM=fda2c8b6b606ff199ea8a6f019008aa8272b7c349cb9312ccd5944153509503a \
+    SPACE_SHA1=cd6fefa63317a7a6effb61a1c5aee634ebe2ca05 \
+    PROSQLITE_SHA1=a1d915d07933ece27ea5fd68f07c83d10583e7a0 \
+    ROCKSDB_SHA1=29eaee6fcdb6dce690ed187ef68b80ee94739412 \
+    HDT_SHA1=e0a0eff87fc3318434cb493690c570e1255ed30e \
+    RSERVE_CLIENT_SHA1=72838bbfa3976a83d19fb38bdae04378e30f4b0d && \
+    BUILD_DEPS='make gcc g++ wget git autoconf libarchive-dev libgmp-dev libossp-uuid-dev libpcre3-dev libreadline-dev libedit-dev libssl-dev zlib1g-dev libdb-dev unixodbc-dev libsqlite3-dev librocksdb-dev libserd-dev libraptor2-dev libgeos++-dev libspatialindex-dev' && \
+    apt-get update && apt-get install -y --no-install-recommends $BUILD_DEPS && \
+    mkdir /tmp/src && \
+    cd /tmp/src && \
+    wget http://www.swi-prolog.org/download/devel/src/swipl-$SWIPL_VER.tar.gz && \
+    echo "$SWIPL_CHECKSUM  swipl-$SWIPL_VER.tar.gz" >> swipl-$SWIPL_VER.tar.gz-CHECKSUM && \
+    sha256sum -c swipl-$SWIPL_VER.tar.gz-CHECKSUM && \
+    tar -xzf swipl-$SWIPL_VER.tar.gz && \
+    cd swipl-$SWIPL_VER && \
+    cp build.templ build && \
+    sed -i '/PREFIX=$HOME/c\PREFIX=/swipl' build && \
+    sed -i '/# export DISABLE_PKGS/c\export DISABLE_PKGS="jpl xpce"' build && \
+    sed -i 's/# *\(EXTRA.*--disable-libdirversion\)/\1/' build && \
+    chmod u+x build && ./build && \
+    cd /usr/bin && rm -rf /tmp/src && ln -s /swipl/bin/swipl swipl && \
+    # space addin
+    mkdir -p /swipl/lib/swipl/pack && \
+    cd /swipl/lib/swipl/pack && \
+    git clone https://github.com/JanWielemaker/space.git && \
+    (cd space && git checkout -q ${SPACE_SHA1}) && \
```

Comment:
https://github.com/JanWielemaker/space/archive/cd6fefa63317a7a6effb61a1c5aee634ebe2ca05.tar.gz

:wink: (no need for `git` anywhere in here)

Reply:
No, but notably the `hdt` and `rserve_client` packages do use git submodules that need to be extracted at the right version.  Of course we could change the package to dynamically fetch its dependencies some other way, but if this isn't really needed I'd propose to keep it as is.  If we use git for downloading some of the packages we might just as well use it for the others I assume.